runtime/src/in_mem_accounts_index.rs

```diff
@@ -538,7 +566,7 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
                 InMemAccountsIndex::lock_and_update_slot_list(
                     occupied.get(),
                     (slot, account_info),
-                    None,
+                    None, // should be None because we don't expect a different slot # during index generation
```

Comment:
Do you think some kind of test or (debug?) assert would be valuable for these cases (other one at line 601) to ensure this fn is only called during index generation and that there's no other dirty slot?